client/src/App.jsx

```diff
@@ -256,12 +256,12 @@ class App extends React.Component {
     var urlStateInfo;
     $.ajax({
       method: 'GET',
-      url: `/${productId}/availableDates`,
+      url: `/rooms/${productId}/availableDates`,
```

Comment:
Your API routes and ajax requests don't necessarily need to be updated to include /rooms. By just updating line 244 like you have, the productId will remain the same and your API routes should work as before if nothing else changes.